beprof/curve.py

```diff
@@ -66,6 +66,19 @@ def y_at_x(self, x):
             return x
         return np.interp(x, self.x, self.y, left=np.nan, right=np.nan)
 
+    def change_domain(self, domain):
+
+        #check if new domain includes in the orginal domain
+        if np.max(domain)>np.max(self.x):
+            #maybe some exception here or stderr log ?
+            print('Error1')
+            return self
+        if np.min(domain)<np.min(self.x):
+            print('Error2')
+            return self
+        obj = Curve([[x, self.y_at_x(x)] for x in domain])
```

Comment:
What if your domain would have million of points ?
self.y_at_x calls np.interp which then calls piece of C code.
switching back and forth from python to C million times will be really time consuming

I think it would be better if you get familiar with np.interp first.